internal/server/data/migrations.go

```diff
@@ -1052,3 +1053,19 @@ func makeIdxEmailsProvidersUnique() *migrator.Migration {
 		},
 	}
 }
+
+func deviceFlowAuthRequestsAddUserIDProviderID() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2022-12-03T13:14",
+		Migrate: func(tx migrator.DB) error {
+			_, err := tx.Exec(`
+				ALTER TABLE device_flow_auth_requests
+					DROP COLUMN if exists access_key_id,
+					DROP COLUMN if exists access_key_token,
+					ADD COLUMN IF NOT EXISTS user_id bigint,
+					ADD COLUMN IF NOT EXISTS provider_id bigint;
```

Comment:
What happens to existing device flow records when we run this migration? I think they will break because user_id and provider_id are null.

I guess since these are short lived and not many people are using it right now that it's ok, but still important to know. We should expect any active device flow request to fail when this is deployed to prod. Might be good to try and deploy during off-peak.

Reply:
This is a great catch. I'll add to the migration to expire existing requests immediately so any mid-device-flow users see a message and can restart the flow.

The alternative would be to populate `user_id` and `provider_id` from the access key, but these are short-lived and it would be better that we get the correct provider_id (which is not the case with the transient access key in the current model).